### ISSUE TITLE
Fix failing TestBytecodeActionEvaluation test

### ISSUE THREAD

**Issue by PaulSnow** [DTRules/DTRules#424]
## Summary

`go test ./...` has one failing test: `TestBytecodeActionEvaluation` in `go/pkg/dtrules/interpreter/state_test.go` (line 983). The assertion "Expected error for unbalanced stack" fails.

## Steps to reproduce

```bash
cd go
go test ./pkg/dtrules/interpreter/ -run TestBytecodeActionEvaluation -v
```

## Expected

Test should pass — the bytecode action evaluation should detect and report unbalanced stack states.

## Notes

This is a unit test issue unrelated to sample projects or integration tests. All other tests pass.
